meson.build

```diff
@@ -309,6 +310,23 @@ if meson.get_compiler('cpp').compiles(
         description : 'Define to 1 if you have the `pubsetbuf` function.')
 endif
 
+
+
+# checking data types
+#============================================================================
+
+
+dirent_h_prefix = '''
+  #include <sys/types.h>
+  #include <dirent.h>
+'''
+
+# Code taken from glib, thanks!
+if has_dirent_h and meson.get_compiler('cpp').has_member('struct dirent', 'd_type', prefix: dirent_h_prefix)
+  define = 'HAVE_STRUCT_@0@_@1@'.format('dirent'.to_upper(), 'd_type'.underscorify().to_upper())
+  config_h.set(define, 1)
+endif
+
```

Comment:
this doesnt quite follow the code style for the rest of the build files, and it doesnt explain whats happening here. theres also no description for the config header

Reply:
So I think the header checks should go *after* the dependency checks, so the headers checks and use the dependencies (good for weird dirs, etc.)

I didn't want to put that much code motion in this PR, but maybe you rather I do for stylistic consistency?

Reply:
mmm good point, maybe re order things in a housecleaning PR later

yep! the high verbosity of the files was purposeful, and id like to see it maintained into the future as much as possible

Reply:
This is still not very clear, it should really be restructured.

Reply:
better :)